### DISCUSSION TITLE
Async Versions of Accounts Functions?

### DISCUSSION THREAD

**Discussion by VikR1000** [meteor/meteor#12296] [Feature Requests]
I have to call some async functions from inside `Accounts.onCreateUser` and `Accounts.emailTemplates.verifyEmail`.

Up until now I have always done it using Meteor's `Promise.await()`.

But with `Promise.await()` going away, I need another way to do that.

I **think** I need an async version of these Accounts functions now.

Is that correct?

**Comment by perbergland** [comment]
@denihs please add a topic on the Fibers board for the work of asyncifying the entire accounts package.

All methods that directly or indirectly touch on mongodb collections for find, update, insert need to be async (e.g. all low level token methods and its callees)
All callbacks that can be registered (onLogin etc) need to be executed with an await of their result since the callback may be async itself (e.g. to use mongodb methods).
This will also impact many accounts addon packages but that work seems to have started already.

I expect that all existing methods will be kept and XAsync versions added

**Reply by denihs** [reply]
[Done](https://github.com/orgs/meteor/projects/8/)! Thank you.
